pkg/workflows/wasm/host/wasip1.go

```diff
@@ -41,9 +43,41 @@ func newWasiLinker(engine *wasmtime.Engine) (*wasmtime.Linker, error) {
 		return nil, err
 	}
 
+	err = linker.FuncWrap(
+		"wasi_snapshot_preview1",
+		"random_get",
+		randomGet,
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	return linker, nil
 }
 
+// randomGet overrides the random_get function in the WASI API and fixes the random source with
+// a hardcoded seed via insecure randomness.  This is to ensure that workflows are deterministic.
+// Randomness should not be used in workflows.
+func randomGet(caller *wasmtime.Caller, buf, bufLen int32) int32 {
+	var (
+		// Fix the random source with a hardcoded seed
+		randSource = rand.New(rand.NewSource(42)) //nolint:gosec
```

Comment:
Shouldn't the randSource be tied to the execution and not the call?  Otherwise, all random numbers will always be the same.

Also, it may be beneficial to use the execution ID at that point so that different runs can have different values.

Reply:
We won't necessarily have an execution ID @nolag 

IMO we should just freeze the randomness and make that clear to users -- do you see any legitimate uses of randomness in workflows?

Reply:
We can make execution id 0, or use the workflow ID during DAG creation.  Is there any other point that we wouldn't have one?

NFT projects often use offline random for packs.  They give a hash of the cards you receive as the pack's ID so that you know it was created before it's purchased and they can't just give you any cards afterwards. 

I haven't seen any use cases here, but I'm worried that just because I haven't seen it doesn't mean there's no other legitimate use.